packages/react-ai/src/components/AIConversation/views/default/MessageList.tsx

```diff
@@ -33,6 +34,35 @@ const MessageMeta = ({ message }: { message: ConversationMessage }) => {
   );
 };
 
+const LoadingMessage = () => {
+  const avatars = React.useContext(AvatarsContext);
+  const variant = React.useContext(MessageVariantContext);
+  const avatar = avatars?.ai;
+
+  return (
+    <View
+      className={classNames(
+        ComponentClassName.AIConversationMessage,
+        classNameModifier(ComponentClassName.AIConversationMessage, variant),
+        classNameModifier(ComponentClassName.AIConversationMessage, 'assistant')
+      )}
+    >
+      <View className={ComponentClassName.AIConversationMessageAvatar}>
+        <Avatar isLoading>{avatar?.avatar}</Avatar>
```

Comment:
We could, but I don't think we should. If you look at the gif the loading avatar shows the avatar icon and the circular loader so your could would need to include both a Loader component and the avatar for the loading state plus some CSS to get it positioned correctly. I think an isLoading prop for an Avatar primitive makes sense in general too.

Reply:
I do somewhat think the `isLoading` prop is very specific to our use case, but maybe that's not an issue since we're the primary user of this primitive right now

Reply:
Ultimately feel coupling loading state/rendering logic inside a UI primitive leads to long term complications with extension of the primitive. Adding `isLoading` to `Avatar` feels like a bit of a foot gun, since as a consumer i now need to make sure that I'm removing `isLoading`/setting to `false` if my avatar is loaded, in addition to the mental overhead of the existing behavior of the `src` and `children` props.

 In this case providing a loading component would just require passing a `className` to our existing `Loader` component. IMO the `Loader` itself should just work out of the box for use cases where it is provided as a placeholder.

Not blocking, but do feel like we gain more from composition and lose flexibility by adding behavioral `props`